src/backend/src/lib.rs

```diff
@@ -154,36 +155,63 @@ pub struct Config {
     pub ecdsa_key_name: String,
     // A list of allowed callers to restrict access to endpoints that do not particularly check or use the caller()
     pub allowed_callers: Vec<Principal>,
+    pub supported_credentials: Option<Vec<SupportedCredential>>,
+    /// Root of trust for checking canister signatures.
+    pub ic_root_key_raw: Option<Vec<u8>>,
+}
+
+fn update_config(arg: InitArg) {
+    mutate_state(|state| {
+        let ic_root_key_raw = match arg
+            .ic_root_key_der
+            .map(|der_key| extract_raw_root_pk_from_der(&der_key))
+        {
+            Some(Ok(root_key)) => Some(root_key),
+            Some(Err(_)) => None,
```

Comment:
It's an acceptable error - i.e. if it fails it's ok to let the init and upgrade of the canister succeed? How do we notice if this fails?

Reply:
Mhm, I don't know. My default is always to keep running code. But maybe in this case we want to panic and skip the upgrade. I'll add a `panic` instead.